File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/CCPADataSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.ccpa.CCPAData

internal fun CCPAData.serialize(): Any {
    return mapOf(
        "version" to version,
        "noticeGiven" to noticeGiven,
        "optedOut" to optedOut,
        "lspact" to lspact
    )
}

File: ios/Classes/Serializer/CCPADataSerializer.swift
import Usercentrics
import Flutter

extension CCPAData {
    func serialize() -> Any {
        return [
            "version": self.version,
            "noticeGiven": Bool(from: self.noticeGiven) as Any,
            "optedOut": Bool(from: self.optedOut) as Any,
            "lspact": Bool(from: self.lspact) as Any,
            "uspString": uspString,
        ]
    }
}
